### PR TITLE
Document instanced pipelines

### PR DESCRIPTION
**Pull request by aoldershaw**
Probably not complete, but meant as a first pass to give some motivation and indicate how to use instanced pipelines

### CODE REVIEW

**Comment by aoldershaw** [comment]
@vito by the way, yesterday I mentioned that the term "instance groups" is just an invented name that's used in the codebase, but that's not entirely true - with concourse/concourse#6433, the search bar got a new `group:` filter that allows you to both filter by the shared name of instanced pipelines (or the instance group name)

The difference between the `group:` filter and just regular search by pipeline name is that in addition to filtering by name, when a `group:` filter is applied, the dashboard drills into instance groups (or expands the instance groups) to view the individual instanced pipeline cards (in fact, clicking an instance group is just a shorthand for applying a `team:` and `group:` filter, i.e. expand just this one instance group)

**Comment by vito** [comment]
@aoldershaw Gotcha, makes sense. I think there are just a couple things about the name that throw me off, but I couldn't quite pinpoint it yet when we were chatting.

The main thing that gives me pause about the term 'instance groups' is that it raises the question as to what an 'instance' is. I wonder if 'pipeline groups' would be clearer? Or maybe we could just spell it out as 'pipeline instance groups'?

One concern is that we already have something that could be called 'pipeline groups', i.e. [`groups:`](https://concourse-ci.org/pipelines.html#schema.pipeline.groups). So maybe that's not a good idea. Maybe 'pipeline sets,' but that could be confusing with `set-pipeline`. 😆 There's 'pipeline matrix', but no one wants to deal with the plural form, and it's a bit overloaded (I don't really think of our set of release pipelines as a 'matrix' for some reason).

We could go full circle and say 'pipeline spaces', but that might make it sound like a loftier abstraction than it really is. (e.g. namespacing)

I think this is kind of an oversight with the RFC - it says how to define each instance, but it doesn't provide a mental model or useful terminology for the resulting set of them which are associated by name.

Just thinking out loud at this point - seems like this is pretty prominent UX that we may want to suss out and make sure we're representing everything as clearly as possible before we remove the 'experimental' label.

(/cc @matthewpereira)

File: lit/docs/pipelines/instance-groups.lit [hunk 1]
```diff
@@ -0,0 +1,212 @@
+\title{Grouping Pipelines}{instance-groups}
```
Comment (vito) [inline lit/docs/pipelines/instance-groups.lit:1]: While we work out a good name for this, can we have a stable tag like `pipeline-instances` or `instanced-pipelines`? That way we can at least start linking to it without breaking the URL later.

**Review by vito — CHANGES_REQUESTED** [review]
Just a few tweaks, I think we can get this out with v7.0 so long as we land on a stable URL for it.

File: lit/docs/jobs/steps.lit [hunk 2]
```diff
@@ -641,8 +641,72 @@ by configuring \reference{schema.step.on_failure} or
...
+        }
+      }
+
       \optional-attribute{vars}{vars}{
-        A map of template variables to pass to the pipeline config.
+        A map of template variables to pass to the pipeline config. Unlike
+        \reference{schema.step.set-pipeline-step.instance_vars}{\code{instance_vars}},
+        \code{vars} are solely used to for
+        \reference{pipeline-static-vars}{interpolation}, and do not become a part of
+        the pipeline's identity
```
[the remark this answers is not in the document]
Comment (aoldershaw) [reply lit/docs/jobs/steps.lit]: Changed to `identifier` which feels a little more concrete than `identity`

**Review by vito — APPROVED** [review]
lgtm!

**Pull request merged**